### PR TITLE
Add padding support to TextInput

### PR DESCRIPTION
**Pull request by furkee**
Reason I need this padding is when TextInput's height is set to be bigger than some certain height, both input text and placeholder text will stay at top even if I set all of textAlign, AlignItems, justifyContent, alignSelf to center(tried them on parent view, on TextInput with all possible configuration). With padding, I was able to move them to center vertically.

Horizontal centering is a bit of issue as well, see issue #793 for that.

### CODE REVIEW

**Comment by rozele** [comment]
Thanks @furkee. The padding needs to be set in the `ReactTextInputShadowNode` instance so it can be incorporated into the `Measure` function, and then it needs to feed through the `UpdateExtraData` ViewManager method (https://github.com/Microsoft/react-native-windows/blob/master/ReactWindows/ReactNative/Views/TextInput/ReactTextInputManager.cs#L424-L435).

**Comment by furkee** [comment]
Thanks for your quick reply @rozele 

How about now?

File: ReactWindows/ReactNative.Shared/Views/TextInput/ReactPasswordBoxShadowNode.cs [hunk 1]
```diff
@@ -179,6 +180,61 @@ public override void OnCollectExtraUpdates(UIViewOperationQueue uiViewOperationQ
         }
 
         /// <summary>
+        /// Computes paddings and sets <see cref="_computedPadding"/> variable so that
```
Comment (matthargett) [inline ReactWindows/ReactNative.Shared/Views/TextInput/ReactPasswordBoxShadowNode.cs]: Not sure how @rozele feels, but I don't see a need to add xml documentation for private methods, nevermind a short one like this. On a side note, even if this was a public method, mentioning a private field detail wouldn't be advised IMO.

File: ReactWindows/ReactNative.Shared/Views/TextInput/ReactTextInputShadowNode.cs [hunk 2]
```diff
@@ -243,14 +243,58 @@ public override void OnCollectExtraUpdates(UIViewOperationQueue uiViewOperationQ
         }
 
         /// <summary>
+        /// Computes paddings and sets <see cref="_computedPadding"/> variable so that
```
Comment (matthargett) [inline ReactWindows/ReactNative.Shared/Views/TextInput/ReactTextInputShadowNode.cs]: same comment here as above.

File: ReactWindows/ReactNative.Shared/Views/TextInput/ReactPasswordBoxShadowNode.cs (hunk 1 above)
Comment (matthargett) [reply ReactWindows/ReactNative.Shared/Views/TextInput/ReactPasswordBoxShadowNode.cs]: A+ for trying to follow the style in the surrounding code, though! 🥇 :)

**Comment by furkee** [comment]
Makes sense not to mention & document private variables and logic. It is not hard to follow where that computation takes anyways. Updated PR with removed said documentation strings. Thanks for your feedback :)

**Comment by rozele** [comment]
Closing this in favor of #1511
